### ISSUE TITLE
httpauth with ":" into password 

### ISSUE THREAD

**Issue by jjeromee** [Qarj/WebInject#23]
Hi,

I would like to know if I may use httpauth with ":" character into the password or any argument ?

Currently I can't make it work, even if I try with quote.

My httpauth looks like that : 

<httpauth>XXXX:443::XX::PASS</httpauth> where ':PASS' is my password

Best Regards.

Jérôme

**Comment by Qarj** [comment]
Hi Jérôme,

There is no way that is going to work since the colon character `:` is used as a delimiter. It is a rather simple system.

Can you change your arguments so none of them use a colon?

Kind Regards,

Tim

**Comment by Qarj** [comment]
I'll probably change this so that the first character specified will be the delimiter. Unfortunately this will break backwards compatibility, but I cannot think of any other way of doing it.

This means

```
<httpauth>www.fakedomain.com:80:my_realm:foo:welcome</httpauth>
```

will become

```
<httpauth>:www.fakedomain.com:80:my_realm:foo:welcome</httpauth>
```

which also means you could use

```
<httpauth>|www.fakedomain.com|80|my_realm|foo|welcome</httpauth>
```

instead - or indeed other character that does not appear in the 5 fields.

I guess I could make it so that if it found exactly 5 colons it would work the old way, but this is a bit hacky.

**Comment by jjeromee** [comment]
Hi Tim,

I found another solution which is currently working :

<testvar varname="LOGIN">mylogin</testvar>
<testvar varname="PASS">mypass</testvar>
<testvar varname="LOGON">https://${LOGIN}:${PASS}@${IP01}</testvar<https://$%7bLOGIN%7d:$%7bPASS%7d@$%7bIP01%7d%3c/testvar>>

It’s work even with the “:” character in the PASS variable.

Best Regards

Jérôme

De : Tim Buckland [mailto:notifications@github.com]
Envoyé : jeudi 3 novembre 2016 08:50
À : Qarj/WebInject WebInject@noreply.github.com
Cc : Jérôme FLEURANCE jerome.fleurance@kleegroup.com; Author author@noreply.github.com
Objet : Re: [Qarj/WebInject] httpauth with ":" into password (#23)

I'll probably change this so that the first character specified will be the delimiter. Unfortunately this will break backwards compatibility, but I cannot think of any other way of doing it.

This means

<httpauth>www.fakedomain.com:80:my_realm:foo:welcome</httpauth<http://www.fakedomain.com:80:my_realm:foo:welcome%3c/httpauth>>

will become

<httpauth>:www.fakedomain.com:80:my_realm:foo:welcome</httpauth>

which also means you could use

<httpauth>|www.fakedomain.com|80|my_realm|foo|welcome</httpauth>

instead - or indeed other character that does not appear in the 5 fields.

I guess I could make it so that if it found exactly 5 colons it would work the old way, but this is a bit hacky.

**Comment by Qarj** [comment]
Nice one! Glad you found a workaround.. I'll put the other fix in later when I get a chance.

Regards,

Tim

**Comment by Qarj** [comment]
Fixed with WebInject 2.3.0
